R/coh_data_wrangling.R

```diff
@@ -498,25 +509,36 @@ get_immunization_dose <- function(data,
     vacc_date_col,
     min.len = 1L
   )
-  checkmate::assert_number(
-    immunization_delay,
-    lower = 0, finite = TRUE
+  stopifnot(
+    "Please provide a non-null integer number greater or equal than 0
+    in `outcome_delay`. Use round(`immunization_delay`,0)" =
+      checkmate::test_integerish(immunization_delay, lower = 0, null.ok = FALSE)
   )
   checkmate::assert_names(
     colnames(data),
     must.include = c(vacc_date_col, immunization_date_col)
   )
+  checkmate::assert_date(
+    data[[immunization_date_col]]
+  )
+  # The function can receive one or more column names in vacc_date_col
+  vacc_date_col_ <- vacc_date_col # hard coded to return right error message
+  for (vacc_date_col in c(vacc_date_col_)) {
+    checkmate::assert_date(
+      data[[vacc_date_col]]
+    )
+  }
 
   # calculate the expected date of immunizing vaccination
   data$delta_imm <- data[[immunization_date_col]] - immunization_delay
 
   # get the first dose corresponding to immunization date - delay
-  dose_number <- apply(data[, c(vacc_date_col, "delta_imm")], 1, function(x) {
+  dose_number <- apply(data[, c(vacc_date_col_, "delta_imm")], 1, function(x) {
     which(x == x[length(x)])[1] # hard coded to get first value
   })
 
   # get names of the vaccination columns corresponding to the dose
-  return(vacc_date_col[dose_number])
+  return(vacc_date_col_[dose_number])
```

Comment:
See suggestion above - these changes could be reverted.

Reply:
I suggest keeping them, based on the previous comment.